lib/Factory/AbstractApiClientFactory.php

```diff
@@ -37,10 +38,20 @@
     public function createTokenManager(
         $baseUri,
         ApiCredentials $credentials,
-        CacheItemPoolInterface $cacheItemPool = null
+        CacheItemPoolInterface $cacheItemPool = null,
+        array $middlewares = []
     ) {
+        $stack = HandlerStack::create();
```

Comment:
HandlerStack is a Guzzle6 specific class - unless Guzzle5 support has been dropped completely, my suggestion is to simply take the argument in the generic abstract factory and in the Guzzle6 implementation one override the createTokenManager